_pages/dev/session-management.md

```diff
@@ -0,0 +1,194 @@
+
+## User Authentication
+
+This section takes a detailed look at the `UserAuthModule`, where most of the changes to session management were made.
+
+The following diagram shows how the `UserAuthModule` works under the hood:
+
+![UserAuthModule]({{ site.baseurl }}/assets/images/session-management/all.svg)
+
+The `UserAuthModule` module is responsible for the following:
+
+- performing authentication flow
+- storing tokens and user identifiers
+- adding access tokens for user calls
+- recovering from API auth errors (for example, refreshing tokens when access tokens expire)
+- persisting the tokens in browser storage
+
+Each of these steps is described in more detail in the following sections.
+
+### Authentication Flow
+
+Authenticating users is the main responsibility of this module. Previous versions of Spartacus used custom code to provide support for the Resource Owner Password Flow, but OAuth specifies additional flows that can be used in web applications. To support these additional flows, Spartacus 3.0 no longer uses its custom code for the Resource Owner Password Flow, and instead relies on the third party `angular-oauth2-oidc` library that is built for this purpose, and is also well-tested and certified.
+
+The services that are involved in authenticating users are highlighted in the following diagram:
+
+![Authentication flow]({{ site.baseurl }}/assets/images/session-management/auth.svg)
+
+Authentication starts from the `AuthService` facade, where you initialize the process by calling one of the following login methods:
+
+- `loginWithCredentials` for the Resource Owner Password Flow
+- `loginWithRedirect` for the Implicit Flow or the Authorization Code Flow
+
+The login method then interacts with the `angular-oauth2-oidc` library. However, this interaction always goes through the `OAuthLibWrapperService`, which is a layer for isolating external libraries from Spartacus code.
+
+**Note:** If you want to access methods from a library that Spartacus does not expose, you should extend the `OAuthLibWrapperService` class.
+
+The `OAuthLibWrapperService` class is also responsible for configuring the library. It uses the `AuthConfigService` to access the `AuthConfig` configuration provided in the application.
+
+It is recommended that you always use the `AuthConfigService` if you need to read anything from the `AuthConfig`. This service is able to provide sensible defaults when some parts of the configuration are missing, and it gives you a few utilities out of the box, including one that detects OAuth flow based on your configuration.
+
+The `OAuthLibWrapperService` is a relatively thin layer that provides a lot of flexibility with the library, and also provides the `authentication.OAuthLibConfig` configuration option, which allows you to pass any configuration option to the library. Client and endpoint configurations are available in the top-level `authentication` configuration.
```

Comment:
```suggestion
The `OAuthLibWrapperService` is a relatively thin layer that provides a lot of flexibility with the library. We also provides the `authentication.OAuthLibConfig` configuration option, which allows you to pass any configuration option to the library. Client and endpoint configurations are available in the top-level `authentication` configuration.
```

The service does not provide this option and I understood previous version like this is the service responsibility.

Reply:
👍